qualang_tools/video_mode/README.md

```diff
@@ -0,0 +1,184 @@
+    ...
+    with infinite_loop():
+        video_mode.load_parameters()
+        ...
+```
+
+### Accessing QUA variables from ```VideoMode``` within the program
+The QUA variables declared in the program through the ```VideoMode.declare_variables()``` can be accessed from the
+through the ```video_mode["param_name"]``` syntax. This enables you to choose where your QUA variable should land in 
+your program. For instance, if you want to use the same QUA variable for two different pulse parameters, you can do the following:
+```
+with program() as video_mode_prog:
+    video_mode.declare_variables()
+    
+    with infinite_loop():
+        video_mode.load_parameters()
+        
+        play("my_pulse"*amp(video_mode["amp_param"]), "qe1")
+        play("my_other_pulse"*amp(video_mode["amp_param"]), "qe2")
+```
+
+Note that you can also access to the full list of QUA variables declared in the program through the ```video_mode.variables``` attribute.
+
+Alternatively, you can also access the QUA variables later in the code as follows:
+```
+with program() as video_mode_prog:
+    amp_param1, amp_param2, ... = video_mode.declare_variables()
+    
+    with infinite_loop():
+        video_mode.load_parameters()
+        
+        play("my_pulse"*amp(amp_param1, "qe1")
+        play("my_other_pulse"*amp(amp_param2), "qe2")
+```
+## Outside the QUA program
+The ```VideoMode``` class has only one ```execute```method to be called outside the QUA program scope, which is a simple wrapper of ```qm.execute()``` method.
+It will start the video mode execution by creating a new thread doing a continuous query of input parameters from the user, while the program is executed and possible live data plotting is on.
+Note that the QUA program runs in the background and the user can still interact with the Python environment while the program is running.
+
+## Example
+Let us consider the following example where we want to update dynamically the amplitude and phase of a pulse.
+We first declare the ```VideoMode``` instance and the parameters dictionary:
+```
+parameters_dict = {
+    'amp': 0.5,
+    'phase': 0.0
+}
+video_mode = VideoMode(qm, parameters_dict)
+```
+We then declare the QUA program as follows:
+```
+with program() as video_mode_prog:
+    video_mode.declare_variables()
+    
+    with infinite_loop():
+        video_mode.load_parameters()
+        
+        play("my_pulse"*amp(video_mode["amp"]), "qe1")
+        frame_rotation(video_mode["phase"], "qe1")
+```
+
+Or an equivalent version:
+```
+with program() as video_mode_prog:
+    amp_, phase = video_mode.declare_variables()
+    
+    with infinite_loop():
+        video_mode.load_parameters()
+        
+        play("my_pulse"*amp(amp_), "qe1")
+        frame_rotation(phase, "qe1")
+```
+Finally, we start the video mode execution:
+```
+video_mode.execute(video_mode_prog)
+```
+The user can then interact with the program by typing the following in the Python console:
+```
+amp=0.8
```

Comment:
This is not 100% clear to me
Does it work when there isn't a console?

Any, it's missing from the examples...

Reply:
It works in a notebook (example is available there), in a terminal, and in the console. The whole idea is that the user should have the ability to type some input parameters with the keyboard while the program is running (like a prompt). Could you specify what you mean by missing from the examples?

Reply:
The notebook wasn't around when I wrote this.

Reply:
I would add: 
At this point, the python console accepts specific inputs, type `help` to see the full list. For example, to change the variables, type: